src/query/api/v1/handler/placement/add.go

```diff
@@ -106,10 +122,36 @@ func (h *AddHandler) Add(
 		return nil, err
 	}
 
+	if !req.Force {
+		curPlacement, _, err := service.Placement()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := validateNoInitializing(curPlacement); err != nil {
+			return nil, err
+		}
+	}
+
 	newPlacement, _, err := service.AddInstances(instances)
 	if err != nil {
 		return nil, err
 	}
 
 	return newPlacement, nil
 }
+
+func validateNoInitializing(p placement.Placement) error {
+	for _, inst := range p.Instances() {
+		total := inst.Shards().NumShards()
```

Comment:
We have some similar functions in m3cluster [here](https://github.com/m3db/m3cluster/blob/master/placement/placement.go#L558), maybe move this logic there as well?